source/LiteralModel.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <mariana-trench/EventLogger.h>
+#include <mariana-trench/LiteralModel.h>
+#include <mariana-trench/Positions.h>
+
+namespace marianatrench {
+
+namespace {
+
+class LiteralModelConsistencyError {
+ public:
+  static void raise(const std::string& what) {
+    ERROR(1, "Literal Model Consistency Error: {}", what);
+    EventLogger::log_event("regex_model_consistency_error", what);
+  }
+};
+
+}
+
+static void check_taint_config_consistency(const std::string& pattern, const TaintConfig& frame) {
+  if (frame.kind() == nullptr) {
+    LiteralModelConsistencyError::raise(fmt::format(
+        "Model for regex `{}` must have a source kind.", pattern));
+  }
+}
+
+static Taint create_taint(const std::string& pattern, const std::vector<TaintConfig>& sources) {
+  Taint all_sources;
+  for (const auto& source : sources) {
+    mt_assert(source.is_leaf());
+    check_taint_config_consistency(pattern, source);
+    all_sources.join_with(Taint{std::move(source)});
```

Comment:
You are using `std::move` on a const ref, that does nothing. might as well remove that `std::move`.

Reply:
Thanks, I changed this to a const ref when I migrated to re2 and I missed the move.